array-methods/reduce/src/__tests__/1_reduce_exercises.test.js

```diff
@@ -119,10 +121,17 @@ We need an array of all savedItemIds from our current customers.
 🛠️ an array of unique savedItemIds across all customers. 
 
 🚨 No duplicate savedItemIds allowed!
-`
+`;
 
 const getSavedItems = customers => {
-  // Your solution here
+  return customers.reduce((uniqueIds, customer) => {
+    customer.savedItemIds.forEach(savedItemId => {
+      if (!uniqueIds.includes(savedItemId)) {
+        uniqueIds = [...uniqueIds, savedItemId];
```

Comment:
```suggestion
        uniqueIds.push(savedItemId);
```
I like this solution better than their fancy-pants use of `Set` which seems very memory intensive and is much less clear in its intent. I'd suggest using a simple `.push` here over the `...spread` because the latter will create a shallow clone of `uniqueIds`, which takes for more memory than necessary for this simple operation.